### PR TITLE
Minimal pipeline main

### PR DESCRIPTION
**Pull request by jefeweisen**
Illustration of how to:
- set up a Pipeline
- set an output directory

To run from the command line:
  sbt compile "runMain org.allenai.pipeline.examples.MinimalSamplePipeline --dry-run"

Note that
  this works:       pipeline.runOnly("MinimalSamplePipeline", copy)
  but this does not:       pipeline.run("MinimalSamplePipeline")

Since dryRun works the second way, it doesn't work either.

### CODE REVIEW

File: src/main/scala/org/allenai/pipeline/examples/MinimalSamplePipeline.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import org.allenai.pipeline.ExternalProcess.{OutputFileToken, InputFileToken}
+import org.allenai.pipeline.IoHelpers._
+import org.allenai.pipeline._
+import org.apache.commons.io.FileUtils
+import spray.json.DefaultJsonProtocol._
+
+import scala.io.Source
+import scala.reflect.ClassTag
+
+class MinimalSamplePipeline extends Pipeline {
```
Comment (rodneykinney) [inline src/main/scala/org/allenai/pipeline/examples/MinimalSamplePipeline.scala:16]: This class doesn't need to both inherit from Pipeline and instantiate a Pipeline.  It should just instantiate one. The `tryCreateArtifact` method can be deleted.

File: src/main/scala/org/allenai/pipeline/examples/MinimalSamplePipeline.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      _.println("Some data")
+    }
+    val inputArtifact = new FileArtifact(inputFile)
+    val outputArtifact = new FileArtifact(outputFile)
+
+    val copy = ExternalProcess("cp", InputFileToken("input"), OutputFileToken("output"))(
+      inputs = Map("input" -> Read.fromArtifact(StreamIo, inputArtifact))
+    )
+      .outputs("output").persisted(StreamIo, outputArtifact)
+    copy.get
```
Comment (rodneykinney) [inline src/main/scala/org/allenai/pipeline/examples/MinimalSamplePipeline.scala:46]: Shouldn't call `get` here.  `pipeline.run()` calls `get` on everything that it's persisted.  Instead:

```
val copy = ExternalProcess("cp", InputFileToken("input"), OutputFileToken("output"))(
  inputs = Map("input" -> Read.fromArtifact(StreamIo, inputArtifact))
)
pipeline.persist(copy, StreamIo)
```

File: src/main/scala/org/allenai/pipeline/examples/MinimalSamplePipeline.scala [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    // END from: it should "read input files":
+
+    if(isDryRun)
+      // TODO: note, this version is currently broken
+      pipeline.dryRun(new File(dir, "dry-run-output"),"MinimalSamplePipeline")
+    else
+      pipeline.runOnly("MinimalSamplePipeline", copy)
```
Comment (rodneykinney) [inline src/main/scala/org/allenai/pipeline/examples/MinimalSamplePipeline.scala:53]: I think `run()` is more appropriate here. The `runOnly` method is intended for when you don't want to compute any non-existent upstream resources.

File: src/main/scala/org/allenai/pipeline/examples/MinimalSamplePipeline.scala [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+object MinimalSamplePipeline {
+  def main(args: Array[String]): Unit = {
+    val isDryRun = args.contains("--dry-run")
+    val that = new MinimalSamplePipeline()
+    try {
+      that.run(isDryRun)
+    }
+    finally {
```
Comment (rodneykinney) [inline src/main/scala/org/allenai/pipeline/examples/MinimalSamplePipeline.scala:65]: What's the point of this `finally`?

**Comment by jefeweisen** [comment]
Will resend.

**Pull request closed without merging**